File: android/src/main/kotlin/com/syncrhrohealth/bandpass_ble_connection/BandpassBleConnectionPlugin.kt
package com.syncrhrohealth.bandpass_ble_connection

import com.syncrhrohealth.bandpass_ble_connection.plugin.devicecore.event.DeviceCoreEventPlugin
import com.syncrhrohealth.bandpass_ble_connection.plugin.devicecore.method.DeviceCoreMethodPlugin
import io.flutter.embedding.engine.plugins.FlutterPlugin

/** BandpassBleConnectionPlugin
 *
 *  Primary entry point for our Flutter plugin. This plugin
 *  internally registers two sub-plugins:
 *   - DeviceCoreMethodPlugin (MethodChannel)
 *   - DeviceCoreEventPlugin  (EventChannel)
 * */
class BandpassBleConnectionPlugin : FlutterPlugin {

    private var deviceCoreMethodPlugin: DeviceCoreMethodPlugin? = null
    private var deviceCoreEventPlugin: DeviceCoreEventPlugin? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // Create + attach the sub-plugins
        deviceCoreMethodPlugin = DeviceCoreMethodPlugin().also {
            it.onAttachedToEngine(binding)
        }

        deviceCoreEventPlugin = DeviceCoreEventPlugin().also {
            it.onAttachedToEngine(binding)
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // Detach each sub-plugin
        deviceCoreMethodPlugin?.onDetachedFromEngine(binding)
        deviceCoreEventPlugin?.onDetachedFromEngine(binding)
    }
}

File: ios/Classes/BandpassBleConnectionPlugin.swift
import Flutter
import UIKit

@objc(BandpassBleConnectionPlugin)
public class BandpassBleConnectionPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let messenger = registrar.messenger()

    // Register all plugin channels
    DeviceCoreMethodPlugin.register(with: messenger)  // MethodChannel for device commands
    DeviceCoreEventPlugin.register(with: messenger)   // EventChannel for device events
    BleStatePlugin.register(with: messenger)          // Event+Method channels for BLE state
  }
}
